pkg/cmd/secret/shared/base_repo_test.go

```diff
@@ -0,0 +1,249 @@
+package shared_test
+
+import (
+	"errors"
+	"testing"
+
+	ghContext "github.com/cli/cli/v2/context"
+	"github.com/cli/cli/v2/pkg/cmd/secret/shared"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cli/cli/v2/git"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/internal/prompter"
+)
+
+func TestRequireNoAmbiguityBaseRepoFunc(t *testing.T) {
+	t.Parallel()
+
+	t.Run("succeeds when there is only one remote", func(t *testing.T) {
+		t.Parallel()
```

Comment:
I think this would make a great topic of conversation about testing craftsmanship.